src/heroku/orb.yml

```diff
@@ -1,35 +1,67 @@
 version: 2
 description: "Provides basic functionality for deploying to Heroku"
 
-commands:
-  install:
+executors:
+  default:
+    description:
+      Uses the basic buildpack-deps image, which has the
+      prerequisites for installing heroku's CLI.
     parameters:
-      os:
-        default: “linux”
-        description: The target Operating System for the heroku binary. Must be one of "linux", “darwin”, “win32”.
+      tag:
         type: string
-        # enum types are not yet implemented.
-        # type: enum
-        # enum: ["linux", “darwin”, “win32”]
-      arch:
-        type: string
-        default: "x86"
-        description: The target CPU architecture for the heroku binary. Must be one of “x64”, “x86”, “arm”.
-        # enum ypes are not yet implemented
-        # type: enum
-        # enum: [“x64”, “x86”, “arm”]
+        default: "bionic"
+    docker:
+      - image: buildpack-deps:<< parameters.tag >>
 
+commands:
+  install:
     steps:
       - run:
           name: "Install Heroku CLI, if necessary"
-          # TODO: the assumption of wget, etc. below likely wouldn't work in all cases, especially windows.
           command: |
             if [[ $(command -v heroku) == "" ]]; then
-              wget https://cli-assets.heroku.com/heroku-cli/channels/stable/heroku-cli-{{parameters.os}}-{{parameters.arch}}.tar.gz -O heroku.tar.gz
-              mkdir $HOME/heroku
-              tar -xvzf heroku.tar.gz -C $HOME/heroku --strip-components 1
-              cd $HOME/heroku
-              ./install
+              curl https://cli-assets.heroku.com/install.sh | sh
             else
               echo "Heroku is already installed. No operation was performed."
             fi
+  deploy:
+    parameters:
+      app-name:
+        description:
+          "The name of your Heroku App.
+          For backwards compatibility the literal value
+          `$HEROKU_APP_NAME` is the default, so you can
+          easily use this command by setting an environment
+          variable called HEROKU_APP_NAME"
+        type: string
+        default: $HEROKU_APP_NAME
+      api-key:
```

Comment:
We should remove this IMO - secrets should be set in env vars, not in config.

Reply:
This does that, but creating an opaque, hard-coded dependency on what those variables can be called would make this harder to use and more prone to frustration. Passing the env vars as the values gives both.

Reply:
Giving users a en empty spot in a config file where we expect them to put their API key will cause them to put their API key into config files. This is a very bad precent to set.